tests/test_omegaconf.py

```diff
@@ -381,3 +398,650 @@ def test_is_issubclass() -> None:
     cfg = OmegaConf.structured(ConcretePlugin)
     t = OmegaConf.get_type(cfg)
     assert t is not None and issubclass(t, ConcretePlugin)
+
+
+def test_to_yaml_list() -> None:
+    c = OmegaConf.create(["item1", "item2", dict(key3="value3")])
+    expected = """- item1
+- item2
+- key3: value3
+"""
+    assert expected == OmegaConf.to_yaml(c)
+    assert OmegaConf.create(OmegaConf.to_yaml(c)) == c
+
+
+def test_to_yaml_list_unicode() -> None:
+    c = OmegaConf.create(["item一", "item二", dict(key三="value三")])
+    expected = """- item一
+- item二
+- key三: value三
+"""
+    assert expected == OmegaConf.to_yaml(c)
+    assert OmegaConf.create(OmegaConf.to_yaml(c)) == c
+
+
+def test_to_yaml_strings_float_list() -> None:
+    c = OmegaConf.create(["10e2", "1.0", 1.0])
+    expected = """- '10e2'
+- '1.0'
+- 1.0
+"""
+    assert OmegaConf.to_yaml(c) == expected
+
+
+def test_to_yaml_string_boolean_list() -> None:
+    for t in _utils.YAML_BOOL_TYPES:
+        print(t)
+        c = OmegaConf.create([t, 1])
+        expected = "- '%s'\n- 1\n" % t
+        assert OmegaConf.to_yaml(c) == expected
+
+
+def test_to_yaml_string_int_list() -> None:
+    c = OmegaConf.create(["1", 1])
+    expected = "- '1'\n- 1\n"
+    assert OmegaConf.to_yaml(c) == expected
```

Comment:
I didn't realize we now have a zillion tests for pretrty.
1. create a new dedicated test file for it : test_to_yaml.py
2. use parameterize to reduce the number of tests. for example the two tests here are almost identical and can be easily pameterized. (do this in two diffs so it's easier to review. first diff moves all the to_yaml tests to one file and the second parameterize.
3.  I see a print here. remove it and other prints you may have added in tests.